sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md

```diff
@@ -0,0 +1,163 @@
+
+## Options
+
+A description of each `CheerpX.Linux.create` option with brief examples are given below.
+
+### `mounts`
+
+```ts
+mounts?: MountPointConfiguration[];
+```
+
+This option configures the filesystems that will be available in the CheerpX environment. Each mount point configures a device and specifies where it should be accessible within the virtual filesystem.
+
+Example:
+
+```js
+const cx = await CheerpX.Linux.create({
+	mounts: [
+		{ type: "ext2", path: "/", dev: overlayDevice },
+		{ type: "tree", path: "/app", dev: webDevice },
+	],
+});
+```
+
+### `networkInterface`
+
+```ts
+networkInterface?: NetworkInterface;
+```
+
+This option configures network settings, which allows CheerpX to communicate over networks.
+
+### `activityInterface`
+
+```ts
+activityInterface?: ActivityInterface;
+```
+
+This option allows monitoring of CPU and device activity within CheerpX, useful for handling state changes and resource management.
+
+### `bridgeURL`
+
+```ts
+bridgeURL?: string;
+```
+
+This option specifies the URL of a bridge server for communication between the CheerpX environment and external servers.
+
+Example:
+
+```js
+const cx = await CheerpX.Linux.create({
+	bridgeURL: "https://yourbridgeurl.com/api",
+});
+```
+
+### Device Configuration Options for CheerpX
+
+CheerpX supports various types of devices that can be configured as overlayDevice. Here’s how you can create them:
+
+- HttpBytesDevice (bytes): The default choice for loading filesystem images via HTTP. Suitable for most web-hosted files.
+- CloudDevice (block): Optimized for use with Cloudflare, enhancing performance and reliability through cloud storage solutions.
+- GitHubDevice (split): Ideal for projects integrated with GitHub Actions, allowing direct file loading from GitHub repositories.
+- IDBDevice: Provides persistent local storage using the browser's IndexedDB, perfect for applications requiring data retention across sessions.
+
+Example: Creating an Overlay Device
+
+```js
+const overlayDevice = await CheerpX.OverlayDevice.create(
+	await CheerpX.HttpBytesDevice.create("https://yourserver.com/image.ext2"),
+	await CheerpX.IDBDevice.create("block1"),
+);
+```
+
+### Using `IDBDevice` in Mounts
+
+`CheerpX.IDBDevice` allows you to create a persistent storage device that can be mounted and accessed like a filesystem within the CheerpX environment.
+
+Example: Mounting IDBDevice
+
+```js
+const idbDevice = await CheerpX.IDBDevice.create("dbName");
+
+const cx = await CheerpX.Linux.create({
+	mounts: [{ type: "tree", path: "/files", dev: idbDevice }],
+});
+```
+
+This setup creates a virtual filesystem at `/files` that is backed by IndexedDB.
+
+### `dataDevice.writeFile`
+
+CheerpX also provides a method to write data to files within the mounted devices. This utility is limited to creating or updating files at the root level. Note that this feature is still subject to review.
```

Comment:
writeFile cannot update files, it can only create a new, previously nonexisting file.

You should remove the subject to review portion of this. When the new API is published the feature will remain as is.